### PR TITLE
fix npe when onCompleteFailure happens after reset

### PR DESCRIPTION
**Pull request by lorban**
Fix for https://github.com/jetty/jetty.project/issues/11841#issuecomment-2305847563

### CODE REVIEW

**Review by sbordet — DISMISSED** [review]

**Comment by garydgregory** [comment]
Hello @lorban,

Thank you for working on this! 
Could the TLS failures be false positives?

File: jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpSender.java [hunk 1]
```diff
@@ -310,7 +310,8 @@ private void terminateRequest(HttpExchange exchange, Throwable failure, Result r
 
     protected void reset()
     {
-        contentSender.reset();
+        if (!contentSender.reset())
+            contentSender = new ContentSender();
```
Comment (sbordet) [inline jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpSender.java]: Don't think this is necessary... Why was it changed?

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpSender.java (hunk 1 above)
Comment (sbordet) [reply jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpSender.java]: Ah, because of the changes to reset().

**Review by sbordet — DISMISSED** [review]

**Comment by lorban** [comment]
@garydgregory unfortunately, they're not. As we cannot reproduce the NPE you reported, this is progressing very slowly as we have to be super careful while working on assumptions.

**Comment by garydgregory** [comment]
> @garydgregory unfortunately, they're not. As we cannot reproduce the NPE you reported, this is progressing very slowly as we have to be super careful while working on assumptions.

@lorban 
TY for the update, I wish I had a better reproducer for you than my initial report in https://github.com/jetty/jetty.project/pull/11891! It seems "random".

**Review by sbordet — APPROVED** [review]

**Comment by garydgregory** [comment]
TY @lorban !

I'll pull, build, and start using it locally.

**Comment by garydgregory** [comment]
Hello @lorban 

Using git `commit ed1cadc449bfd711e7fb5d4d409e4eb57be9321b (HEAD -> jetty-12.0.x, origin/jetty-12.0.x, origin/HEAD)`, I just saw:
```java
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.jetty.client.transport.HttpExchange.getRequest()" because "exchange" is null
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at org.eclipse.jetty.client.transport.HttpRequest.send(HttpRequest.java:707)
	at <our code, redacted>
	... 18 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.jetty.client.transport.HttpExchange.getRequest()" because "exchange" is null
	at org.eclipse.jetty.client.transport.HttpSender$ContentSender.process(HttpSender.java:511)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:262)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:243)
	at org.eclipse.jetty.client.transport.HttpSender.send(HttpSender.java:85)
	at org.eclipse.jetty.client.transport.internal.HttpChannelOverHTTP.send(HttpChannelOverHTTP.java:86)
	at org.eclipse.jetty.client.transport.HttpChannel.send(HttpChannel.java:142)
	at org.eclipse.jetty.client.transport.HttpConnection.send(HttpConnection.java:112)
	at org.eclipse.jetty.client.transport.internal.HttpConnectionOverHTTP$Delegate.send(HttpConnectionOverHTTP.java:379)
	at org.eclipse.jetty.client.transport.internal.HttpConnectionOverHTTP.send(HttpConnectionOverHTTP.java:168)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:444)
	at org.eclipse.jetty.client.transport.HttpDestination.process(HttpDestination.java:420)
	at org.eclipse.jetty.client.transport.HttpDestination.process(HttpDestination.java:375)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:358)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:352)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:329)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:308)
	at org.eclipse.jetty.client.transport.HttpRequest.sendAsync(HttpRequest.java:751)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:303)
	at org.eclipse.jetty.client.transport.HttpRequest.send(HttpRequest.java:744)
	at org.eclipse.jetty.client.CompletableResponseListener.send(CompletableResponseListener.java:79)
	... 20 more
```

**Comment by lorban** [comment]
@garydgregory could we please follow up this conversation on https://github.com/jetty/jetty.project/pull/12210?

**Comment by garydgregory** [comment]
> @garydgregory could we please follow up this conversation on #12210?

👍

**Pull request merged**